File: src/api/MoRS.ManagementSystem.API/DTOs/Appointment/CreateAppointmentRequest.cs
using MoRS.ManagementSystem.API.DTOs.AppointmentSchedule;

namespace MoRS.ManagementSystem.API.DTOs.Appointment;

public class CreateAppointmentRequest
{
    public bool IsRepeating { get; set; }
    public CreateAppointmentScheduleRequest? AppointmentSchedule { get; set; }
    public int RoomId { get; set; }
    public int AppointmentTypeId { get; set; }
    public int? BookedByUserId { get; set; }
    public List<int>? AttendeesIds { get; set; }
}

File: src/mobile/lib/models/appointment/create_appointment_request.dart
import '../appointment_schedule/create_appointment_schedule_request.dart';

class CreateAppointmentRequest {
  final bool isRepeating;
  final CreateAppointmentScheduleRequest appointmentSchedule;
  final int roomId;
  final int appointmentTypeId;
  final int bookedByUserId;
  final List<int>? attendeesIds;

  CreateAppointmentRequest({
    required this.isRepeating,
    required this.appointmentSchedule,
    required this.roomId,
    required this.appointmentTypeId,
    required this.bookedByUserId,
    this.attendeesIds,
  });

  factory CreateAppointmentRequest.fromJson(Map<String, dynamic> json) {
    return CreateAppointmentRequest(
      isRepeating: json['isRepeating'],
      appointmentSchedule: CreateAppointmentScheduleRequest.fromJson(
        json['appointmentSchedule'],
      ),
      roomId: json['roomId'],
      appointmentTypeId: json['appointmentTypeId'],
      bookedByUserId: json['bookedByUserId'],
      attendeesIds: json['attendeesIds'] != null
          ? List<int>.from(json['attendeesIds'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'isRepeating': isRepeating,
      'appointmentSchedule': appointmentSchedule.toJson(),
      'roomId': roomId,
      'appointmentTypeId': appointmentTypeId,
      'bookedByUserId': bookedByUserId,
      'attendeesIds': attendeesIds,
    };
  }
}
